src/main/java/com/conveyal/datatools/common/utils/aws/IAMUtils.java

```diff
@@ -0,0 +1,50 @@
+package com.conveyal.datatools.common.utils.aws;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementClientBuilder;
+import com.amazonaws.services.identitymanagement.model.InstanceProfile;
+import com.amazonaws.services.identitymanagement.model.ListInstanceProfilesResult;
+
+public class IAMUtils {
+    private static final AmazonIdentityManagement DEFAULT_IAM_CLIENT = AmazonIdentityManagementClientBuilder
+        .defaultClient();
+    private static IAMClientManagerImpl IAMClientManager = new IAMClientManagerImpl(DEFAULT_IAM_CLIENT);
+
+    /**
+     * A class that manages the creation of IAM clients.
+     */
+    private static class IAMClientManagerImpl extends AWSClientManager<AmazonIdentityManagement> {
+        public IAMClientManagerImpl(AmazonIdentityManagement defaultClient) {
+            super(defaultClient);
+        }
+
+        @Override
+        public AmazonIdentityManagement buildDefaultClientWithRegion(String region) {
+            return defaultClient;
+        }
+
+        @Override
+        public AmazonIdentityManagement buildCredentialedClientForRoleAndRegion(
+            AWSCredentialsProvider credentials, String region, String role
+        ) {
+            return AmazonIdentityManagementClientBuilder.standard().withCredentials(credentials).build();
```

Comment:
Does IAM not use region? If not, please add comments for any of the methods in this class.

Reply:
Hmm, I'm not sure, but I went ahead and added in the region qualifier just in case in c84b36934f34ac60af86ddf920f5c6a8cc6c7510.